cmd/cone/get_drop_task.go

```diff
@@ -99,6 +101,45 @@ func validateGrantTaskArguments(maxProvisionTime *time.Duration, duration *time.
 	return nil
 }
 
+func getValidJustification(ctx context.Context, v *viper.Viper, justification string) (*string, error) {
+	if strings.TrimSpace(justification) != "" {
+		return &justification, nil
+	}
+
+	if v.GetBool(nonInteractiveFlag) {
+		pterm.Info.Println(justificationInputTip)
+		return nil, errors.New(justificationErrorMessage)
+	}
+
+	input := pterm.DefaultInteractiveTextInput.WithMultiLine(false)
+	firstRun := true
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
+
+		if !firstRun {
+			var err error
+			justification, err = input.Show()
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		firstRun = false
+		if strings.TrimSpace(justification) == "" {
+			pterm.Error.Println(justificationErrorMessage)
+			pterm.Info.Println(justificationInputTip)
+			continue
+		}
+
+		return &justification, nil
+	}
```

Comment:
Now that we've added this flow (multi-input with validation) in more than one place, I think it's valid to extract this into a generic handling function.